Initial commit (#1)
napptive/kubecon-23-oam-kubevela-tutorial: doc/01.install_kubevela.md

# Installing a basic KubeVela development environment

> In this step, we will explore different methods to create a basic KubeVela local environment. The recommended path relies on Kind, while it is also possible to install it with [VelaD](#alternatively-with-velad). Alternatively, you may also try [Napptive](http://try.napptive.com/playground?campaign_source=kubecon23) to deploy simple OAM applications.

Reviewer: ```suggestion
This step will explore different methods to create a basic KubeVela local environment. The recommended path relies on Kind, while it is also possible to install it with [VelaD](#alternatively-with-velad). Alternatively, you may try [Napptive](http://try.napptive.com/kubecon2023) to deploy simple OAM applications.
```

```bash
kubectl --context kind-kubevela apply -f https://raw.githubusercontent.com/kubernetes/ingress-nginx/main/deploy/static/provider/kind/deploy.yaml
```

```bash
kubectl --context kind-kubevela wait --namespace ingress-nginx \
  --for=condition=ready pod \
  --selector=app.kubernetes.io/component=controller \
  --timeout=90s
```

Once everything is installed, we will check the state of the installation with:

```bash
kubectl --context kind-kubevela get nodes
kubectl --context kind-kubevela -n vela-system get pods
```

Finally, let's install the Vela CLI tool. Notice that the tool may require root permissions to copy the binary. For other installation methods, check the [official documentation](https://kubevela.io/docs/installation/kubernetes#install-vela-cli).

```bash
curl -fsSl https://kubevela.net/script/install.sh | bash
```

Once the CLI is available, you should be able to perform OAM related operations such as obtaining the list of available component definitions.

```bash
vela comp
```

Once you no longer require the development cluster, you can delete it with:

```bash
kind delete cluster --name=kubevela
```

## Alternatively with VelaD

> [VelaD](https://github.com/kubevela/velad) is a lightweight tool to deploy KubeVela and that relies on [K3s](https://k3s.io/). The tool is in active development and the resulting cluster may be used for the initial part of the tutorial. It is not recommended for the multi-cluster deployment section as it will require reconfiguring the networking.

Reviewer: ```suggestion
[VelaD](https://github.com/kubevela/velad) is a lightweight tool to deploy KubeVela, and that relies on [K3s](https://k3s.io/). The tool is in active development, and the resulting cluster may be used for the initial part of the tutorial. It is not recommended for the multi-cluster deployment section as it will require reconfiguring the networking.
```

> To install the tool, use the following command. Notice that it may ask for root priviledges to install the binary. Check the [official documentation](https://kubevela.io/docs/installation/standalone#2-install-velad-and-setup-kubevela) for more information.

Reviewer: ```suggestion
To install the tool, use the following command. Notice that it may ask for root privileges to install the binary. Check the [official documentation](https://kubevela.io/docs/installation/standalone#2-install-velad-and-setup-kubevela) for more information.
```
